### ISSUE TITLE
squlite3 issue with Etherpad

### ISSUE THREAD

**Issue by ctwelve** [spantaleev/matrix-docker-ansible-deploy#874]
Hello,

On attempting to enable Etherpad, I run into an issue where the container complains about missing sqlite3:

```
Feb 14 19:46:28 matrix.arkmuse.org systemd[1]: Started Matrix Etherpad.
Feb 14 19:46:29 matrix.arkmuse.org matrix-etherpad[32447]: [2021-02-14 19:46:29.115] [DEBUG] console - Running on Node v10.23.0 (minimum required Node version: 10.13.0)
Feb 14 19:46:29 matrix.arkmuse.org matrix-etherpad[32447]: [2021-02-14 19:46:29.283] [INFO] console - All relative paths will be interpreted relative to the identified Etherpad base dir: /opt/etherpad-lite
Feb 14 19:46:29 matrix.arkmuse.org matrix-etherpad[32447]: [2021-02-14 19:46:29.284] [INFO] console - Random string used for versioning assets: ada4442b
Feb 14 19:46:29 matrix.arkmuse.org matrix-etherpad[32447]: [2021-02-14 19:46:29.287] [INFO] console - settings loaded from: /data/settings.json
Feb 14 19:46:29 matrix.arkmuse.org matrix-etherpad[32447]: [2021-02-14 19:46:29.288] [INFO] console - No credentials file found in /data/credentials.json. Ignoring.
Feb 14 19:46:29 matrix.arkmuse.org matrix-etherpad[32447]: Using skin "colibris" in dir: /opt/etherpad-lite/src/static/skins/colibris
Feb 14 19:46:29 matrix.arkmuse.org matrix-etherpad[32447]: Session key loaded from: /data/sessionkey.json
Feb 14 19:46:29 matrix.arkmuse.org matrix-etherpad[32447]: FATAL: The sqlite dependency could not be loaded. We removed it from the dependencies since it caused problems on several Platforms to compile it. If you still want to use sqlite, do a 'npm install sqlite3' in your etherpad-lite root folder
Feb 14 19:46:29 matrix.arkmuse.org systemd[1]: matrix-etherpad.service: Main process exited, code=exited, status=1/FAILURE
Feb 14 19:46:29 matrix.arkmuse.org systemd[1]: matrix-etherpad.service: Failed with result 'exit-code'.
```

I am not sure what is causing this, but my installation has an external PostgreSQL setup, if that helps...

**Comment by spantaleev** [comment]
Interesting!

Given that you're using an external Postgres, we can't currently set you up on Postgres automatically, so we try to use SQLite for Etherpad and all other services that require a database.

Seems like the Etherpad container image cannot, by default, run on SQLite though. Maybe it's something we've never tested as part of #804. @pushytoxin?

@ctwelve, you can create a new Postgres database and credentials for Etherpad and adjust these variables: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/894679750e8599206362c7cb742e891e1cf2cf53/roles/matrix-etherpad/defaults/main.yml#L32-L48

(in your own `vars.yml` file)

**Comment by ctwelve** [comment]
Oh dear! Is there a guide for migrating all the container databases to Postgres? Did I miss something?

**Comment by spantaleev** [comment]
Most services have always used SQLite. Services being able to use Postgres is a relatively new thing: https://github.com/spantaleev/matrix-docker-ansible-deploy/blob/master/CHANGELOG.md#the-big-move-to-all-on-postgres-potentially-dangerous

By default, all services use Postgres now.. but when external Postgres is used, we can't currently do such things.

You can keep using SQLite for all other services though, at least for now. In the future, should the situation improve, there might be an easier path for running everything on Postgres even when external Postgres is used.

**Comment by pushytoxin** [comment]
It turns out some people didn't like how long it takes to run `npm install` when the database module "ueberdb2" compiled the sqlite3 bindings, so they made the sqlite3 dependency optional.

Unfortunately this makes the Docker image not very good. It also makes the Etherpad documentation misleading to anyone who reads it. I fully expected the Docker image to work with sqlite.

The Docker image has to be built with the additional dependancy, e.g:

    docker build --build-arg ETHERPAD_PLUGINS=sqlite3 -t etherpad:sqlite3 .

This means we should probably add the etherpad image to the list of self-built containers.
We could try to convince the Etherpad devs to include sqlite3 in the Docker image.
But our own documentation should be fixed first.

@ctwelve I saw some mentions of support for a "dirtydb" database, but that's "only for testing".

**Comment by spantaleev** [comment]
I guess this is somewhat fixed (well, avoided) by #883.
